addons/binding/org.openhab.binding.spotify/src/main/java/org/openhab/binding/spotify/internal/api/SpotifyApi.java

```diff
@@ -0,0 +1,293 @@
+
+        return deviceList == null || deviceList.getDevices() == null ? Collections.emptyList()
+                : deviceList.getDevices();
+    }
+
+    /**
+     * @return Returns the playlists of the user.
+     */
+    public List<Playlist> getPlaylists() {
+        final ContentResponse response = request(GET, SPOTIFY_API_URL + "/playlists", "");
+        final Playlists playlists = ModelUtil.gsonInstance().fromJson(response.getContentAsString(), Playlists.class);
+
+        return playlists == null || playlists.getItems() == null ? Collections.emptyList() : playlists.getItems();
+    }
+
+    /**
+     * @return Calls Spotify Api and returns the current playing context of the user or an empty object if no context as
+     *         returned by Spotify
+     */
+    public CurrentlyPlayingContext getPlayerInfo() {
+        final ContentResponse response = requestPlayer(GET, "");
+        final CurrentlyPlayingContext context = ModelUtil.gsonInstance().fromJson(response.getContentAsString(),
+                CurrentlyPlayingContext.class);
+
+        return context == null ? EMPTY_CURRENTLYPLAYINGCONTEXT : context;
+    }
+
+    /**
+     * Calls the Spotify player Web Api with the given method and appends the given url as parameters of the call to
+     * Spotify.
+     *
+     * @param method Http method to perform
+     * @param url url path to call to spotify
+     * @return the response give by Spotify
+     */
+    private ContentResponse requestPlayer(HttpMethod method, String url) {
+        return requestPlayer(method, url, "");
+    }
+
+    /**
+     * Calls the Spotify player Web Api with the given method and appends the given url as parameters of the call to
+     * Spotify.
+     *
+     * @param method Http method to perform
+     * @param url url path to call to spotify
+     * @param requestData data to pass along with the call as content
+     * @return the response give by Spotify
+     */
+    private ContentResponse requestPlayer(HttpMethod method, String url, String requestData) {
+        return request(method, SPOTIFY_API_PLAYER_URL + (url.isEmpty() ? "" : ('/' + url)), requestData);
+    }
+
+    /**
+     * Calls the Spotify Web Api with the given method and given url as parameters of the call to Spotify.
+     *
+     * @param method Http method to perform
+     * @param url url path to call to spotify
+     * @param requestData data to pass along with the call as content
+     * @return the response give by Spotify
+     */
+    private ContentResponse request(HttpMethod method, String url, String requestData) {
+        logger.debug("Request: ({}) {} - {}", method, url, requestData);
+        final Function<HttpClient, Request> call = httpClient -> httpClient.newRequest(url).method(method)
+                .header("Accept", CONTENT_TYPE).content(new StringContentProvider(requestData), CONTENT_TYPE);
+        try {
+            try {
```

Comment:
I think this is not ideal, would it make sense to split up the methods so that we do not have to explicitly nest two try-statements.